python/hail/expr/expression.py

```diff
@@ -1986,7 +1989,10 @@ def __getitem__(self, item):
         :class:`.Expression`
             Struct field.
         """
-        return self._get_field(item)
+        if isinstance(item, str):
+            return self._get_field(item)
+        else:
+            return self._get_field(self.dtype.fields[item].name)
```

Comment:
IndexError on out of bounds int from here is OK, yeah?

Reply:
Yeah, I think so.